WordPress/Sniffs/WP/DeprecatedParametersSniff.php

```diff
@@ -0,0 +1,317 @@
+		'trackback_url' => array(
+			1 => array(
+				'value'   => 'true',
+				'version' => '2.5',
+			),
+		),
+		'wp_install' => array(
+			5 => array(
+				'value'   => '',
+				'version' => '2.6',
+			),
+		),
+		'discover_pingback_server_uri' => array(
+			2 => array(
+				'value'   => '',
+				'version' => '2.7',
+			),
+		),
+		'load_plugin_textdomain' => array(
+			2 => array(
+				'value'   => 'false',
+				'version' => '2.7',
+			),
+		),
+		'wp_get_http_headers' => array(
+			2 => array(
+				'value'   => 'false',
+				'version' => '2.7',
+			)
+		),
+		'safecss_filter_attr' => array(
+			2 => array(
+				'value'   => '',
+				'version' => '2.8.1',
+			)
+		),
+		'wp_get_sidebars_widgets' => array(
+			1 => array(
+				'value'   => 'true',
+				'version' => '2.8.1',
+			),
+		),
+		'get_delete_post_link' => array(
+			2 => array(
+				'value'   => '',
+				'version' => '3.0',
+			)
+		),
+		'is_email' => array(
+			2 => array(
+				'value'   => 'false',
+				'version' => '3.0',
+			),
+		),
+		'update_user_status' => array(
+			4 => array(
+				'value'   => 'null',
+				'version' => '3.0.2',
+			),
+		),
+		'update_blog_option' => array(
+			4 => array(
+				'value'   => 'null',
+				'version' => '3.1',
+			),
+		),
+		'update_blog_status' => array(
+			4 => array(
+				'value'   => 'null',
+				'version' => '3.1',
+			),
+		),
+		'wp_notify_postauthor' => array(
+			2 => array(
+				'value'   => 'null',
+				'version' => '3.8',
+			)
+		),
+		'wp_new_user_notification' => array(
+			2 => array(
+				'value'   => 'null',
+				'version' => '4.3.1',
+			)
+		),
+		'get_wp_title_rss' => array(
+			1 => array(
+				'value'   => '&#8211;',
+				'version' => '4.4',
+			)
+		),
+		'wp_title_rss' => array(
+			1 => array(
+				'value'   => '&#8211;',
+				'version' => '4.4',
+			),
+		),
+		'unregister_setting' => array(
+			4 => array(
+				'value'   => '',
+				'version' => '4.7',
+			),
+		),
+
+	);
+
+	/**
+	 * Process the parameters of a matched function.
+	 *
+	 * @since 0.11.0
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 *
+	 * @return void
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+		foreach ( $this->target_functions[ $matched_content ] as $position => $paramerter_args ) {
+
+			if ( ! isset( $parameters[ $position ] ) ) {
+				continue;
+			}
+
+			$is_dynamic_parameter = $this->phpcsFile->findNext( array( T_VARIABLE, T_STRING ), $parameters[ $position ]['start'], $parameters[ $position ]['end'], false, null, true );
+
+			$matched_parameter = $this->strip_quotes( $parameters[ $position ]['raw'] );
+
+			if ( ! $is_dynamic_parameter && $paramerter_args[ 'value' ] === $matched_parameter ) {
```

Comment:
> If boolean false is expected and string false is passed, according to this check, everything would be fine, while in reality string false would translate to boolean true which would be an invalid value.

This should be fixed now.

> Also, null should always be considered valid as it will just fall through to the parameter default.

I tested this and it did not work for me http://sandbox.onlinephpfunctions.com/code/157fe24a88a2bbf0ec7fc981a56e0a5f2f8f3e18